ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleDBTest.java

```diff
@@ -90,56 +73,59 @@ public void useAppContext() {
 
         }
 
-    }
-
+        mSavedArticles = mArticleDao.getTestableAllArticles();
 
-    private void associateToArticles(List<Article> savedArticles) {
-        for (Article article : savedArticles != null ? savedArticles : new ArrayList<Article>()) {
-            // Create an attachment for each
-            Attachment attachment = new Attachment();
-            attachment.mArticleKey = article.mId;
-            attachment.mAttachmentUrl = "https://i.pinimg.com/originals/b9/9e/a0/b99ea0abc5e6bf5f5c67c1797791cc82.jpg";
-            mAttachmentDao.insertAttachment(attachment);
 
-            // Create Manifest Association
+        for (int i = 0; i < mSavedArticles.size(); i++) {
             ManifestAssociation association = new ManifestAssociation();
-            association.mArticleId = article.mId;
-            association.mManifestId = String.valueOf(article.mId % 3);  // just need to only have three versions
-            association.mManifestName = String.format("Manifest # %s", association.mManifestId);
+            association.mArticleId = mSavedArticles.get(i).mId;
+            association.mManifestName = "Manifest ID " + (i % 3);
+            association.mManifestId = (i % 3) + "";
             mAssociationDao.insertAssociation(association);
+
+            Attachment attachment = new Attachment();
+            attachment.mArticleKey = mSavedArticles.get(i).mId;
+            attachment.mAttachmentUrl =
+                    "https://believeacts2blog.files.wordpress.com/2015/10/image16.jpg";
         }
+
     }
 
-    private void getAllSavedArticles() {
-        mArticleDao.getAllArticles().observe(this, articles -> {
-            associateToArticles(articles);
+    @After
+    public void closeDb() {
+        db.close();
+    }
 
-            //verify that we are getting bat the Articles
-            Boolean allAttachmentsSaved = false;
+    @Test
+    public void verifyGetAllArticles() {
 
-            for (Article article : articles) {
-                if (mAttachmentDao.getAttachmentsByArticle(article.mId) == null) {
-                    allAttachmentsSaved = false;
-                    break;
-                }
+        assert (mSavedArticles.size() > 1);
```

Comment:
you should use the junit assert methods. The built in java `assert` is ignored on android under certain scenarios. Plus the JUnit assert methods allow you to specify a message with the error thrown and give more description messages.